include/libopencm3/stm32/common/rcc_common_all.h

```diff
@@ -67,6 +67,13 @@ bool rcc_is_osc_ready(enum rcc_osc osc);
  */
 void rcc_wait_for_osc_ready(enum rcc_osc osc);
 
+/**
+ * This will return the divisor 1/2/4/8/16/64/128/256/512 which is set as a
+ * 4-bit value, typically used for hpre and other prescalers.
+ * @param div_val  Masked and shifted divider value from register (e.g. RCC_CFGR)
+ */
+uint16_t rcc_prediv_log_skip32_div(uint8_t div_val);
```

Comment:
fwiw, personally I hate this name, and I don't think it needs to be public api.  (obviously I'm not math enough to just go "oh yeah, I need this log function" but I don't think any of your log functions for working out bit values are really api level, they'r einternals to make the library developers day easier)

Reply:
I had it as a static helper in the l0/l4 implementations, and also needed it in the h7, so I figured I'd make it common.  If you don't mind having a few copies of this function running around in separate rcc.c files, I'm fine having the copies.

Reply:
eh, ok then, I guess we can let the optimizers deal with it.

Reply:
renamed `rcc_get_div_from_hpre()` since that's what it's used for :-)